### ISSUE TITLE
Support rspec message expectations

### ISSUE THREAD

**Issue by somebody32** [moonmaster9000/quacky#14]
Hi!

I think that it will be really cool if Quacky could mimic as a real rspec double and support methods like once, twice, ordered etc

Or maybe will be better to extend rspec double class and not replace it completely?

**Comment by benmoss** [comment]
Can you explain this a little more? The only time I've used `once`, `twice`, etc is chained onto `should_receive`, which with RSpec exists on every object.

**Comment by somebody32** [comment]
Yes, because should_receive returns RSpec::Mocks::MessageExpectation object but if you do it on quacky double then you will get Quacky::Double that doesn't implement these messages, so a little example:

without Quacky

``` ruby
class Shop
  def self.populate_store
    RemoteService.request_catalogue!
  end
end

describe Shop do
  let(:remote) { stub_const("RemoteService", double }

  it "populates store" do
    remote.should_receive(:request_catalogue!).once
    described_class.populate_store
  end
end
```

test passes

with quacky:

``` ruby
require 'quacky'

module RemoteInterface
  def request_catalogue!; end
end

class Shop
  def self.populate_store
    RemoteService.request_catalogue!
  end
end

describe Shop do
  let(:remote) { stub_const("RemoteService", Quacky.double(:remote, RemoteInterface)) }

  it "populates store" do
    remote.should_receive(:request_catalogue!).once
    described_class.populate_store
  end
end
```

fails because of unexpected method

Or maybe I'm doing something wrong here?

**Comment by benmoss** [comment]
Ah, I dig. I'll try to poke around and see if I can figure out a solution.

**Comment by benmoss** [comment]
I think the short answer is that as long as we are trying to be agnostic about support for RSpec and MiniTest we won't be able to return actual RSpec doubles.

The long answer is that we could probably use some kind of feature detection to see if the RSpec is available and have a separate code path which includes RSpec's [TestDouble](https://github.com/rspec/rspec-mocks/blob/master/lib/rspec/mocks/test_double.rb) module. That'd be a fair amount of refactoring the way this library works though.
